content/200-concepts/100-components/01-prisma-schema/08-views.mdx

```diff
@@ -0,0 +1,180 @@
+}
+```
+
+## How to add views to your Prisma schema
+
+To add a view to your Prisma schema, use the `view` keyword. For example, start with a `User` model with a linked `Profile`:
+
+<TabbedContent tabs={[<FileWithIcon text="Relational databases" icon="database"/>, <FileWithIcon text="MongoDB" icon="database"/>]}>
+<tab>
+
+```prisma
+model User {
+  id      Int      @id @default(autoincrement())
+  email   String   @unique
+  name    String?
+  profile Profile?
+}
+
+model Profile {
+  id     Int    @id @default(autoincrement())
+  bio    String
+  user   User   @relation(fields: [userId], references: [id])
+  userId Int    @unique
+}
+```
+
+</tab>
+<tab>
+
+```prisma
+model User {
+  id      String   @id @default(auto()) @map("_id") @db.ObjectId
+  email   String   @unique
+  name    String?
+  profile Profile?
+}
+
+model Profile {
+  id     String @id @default(auto()) @map("_id") @db.ObjectId
+  bio    String
+  userId String @unique @db.ObjectId
+  User   User   @relation(fields: [userId], references: [id])
+}
+```
+
+</tab>
+</TabbedContent>
+
+Next, take a `UserInfo` view in the underlying database that combines the `email` and `name` fields from the `User` model and the `bio` field from the `Profile` model.
+
+For a relational database, the SQL statement to create this view is:
+
+```sql
+CREATE VIEW "UserInfo" AS
+    SELECT u.id, email, name, bio
+    FROM "User" u
+    LEFT JOIN "Profile" p ON u.id = p."userId"
+```
+
+For MongoDB, you can [create a view](https://www.mongodb.com/docs/manual/core/views/join-collections-with-view/) with the following command:
+
+```t
+db.createView( "UserInfo", "User", [
+   {
+      $lookup:
+         {
+            from: "Profile",
+            localField: "_id",
+            foreignField: "userId",
+            as: "ProfileData"
+         }
+   },
+   {
+      $project:
+         {
+           _id: 1,
+           email: 1,
+           name: 1,
+           bio: "$ProfileData.bio"
+         }
+   },
+      { $unwind: "$bio" }
+] )
+```
+
+You can represent the `UserInfo` view in your Prisma schema as follows:
+
+<!-- prettier-ignore-start -->
+
+<TabbedContent tabs={[<FileWithIcon text="Relational databases" icon="database"/>, <FileWithIcon text="MongoDB" icon="database"/>]}>
+<tab>
+
+```prisma
+view UserInfo {
+  id    Int    @id
+  email String
+  name  String
+  bio   String
+}
+```
+
+</tab>
+<tab>
+
+```prisma
+view UserInfo {
+  id    String @id @default(auto()) @map("_id") @db.ObjectId
+  email String
+  name  String
+  bio   String
+}
+```
+
+</tab>
+</TabbedContent>
+
+<!-- prettier-ignore-end -->
+
+As with models, a view needs to have a [unique identifier](/concepts/components/prisma-schema/data-model#defining-an-id-field). In relational databases, this can be an `@id` or `@unique` attribute on one field, or an `@@id` or `@@unique` attribute on multiple fields. In MongoDB, this must be an `@id` attribute that maps to the `_id` field in the underlying database with `map(_id)`.
```

Comment:
This is a current and somewhat non-intuitive limitation of our view implementation. We might want to highlight that some way.

Reply:
Is it just the MongoDB part that's unintuitive, or having a unique identifier at all?

Reply:
For users the complete thing. A view does not have a primary key, and also does not define any unique fields - that is done in the underlying tables the query touches. 

Later in Introspection of views we will also not be able to get this information.

Reply:
Thanks. I've just made this into a Limitations section, see comment below